lib/assessment.js

```diff
@@ -29,9 +39,17 @@ var newArray = function newArray(length, defaultValue) {
 // invoking the argument defaultsFunction
 // defaultsFunction should be passed the index of the array element to be set
 var newArray2 = function newArray2(length, defaultsFunction) {
-
+  ary = [];
+  ary.length = length;
```

Comment:
This line will not do what you think it does. In fact, don't ever write this, because it will really mess up the array that you do it to. `.length` is a attribute of an array, not a method; the way that it's kept accurate is that every time an element gets added or removed, the `.length` value gets updated automatically to account for the change. If you manually set `.length` to a different value - say 10 - if you add one more element, `.length` will be equal to 11 _irrespective of how long the array actually is_. See why that might be dangerous?